rclcpp/src/rclcpp/executors/static_single_threaded_executor.cpp

```diff
@@ -175,9 +175,10 @@ StaticSingleThreadedExecutor::execute_ready_executables()
   }
   // Execute all the ready waitables
   for (size_t i = 0; i < entities_collector_->get_number_of_waitables(); ++i) {
-    if (entities_collector_->get_waitable(i)->is_ready(&wait_set_)) {
-      std::shared_ptr<void> shared_ptr;
-      entities_collector_->get_waitable(i)->execute(shared_ptr);
+    auto waitable = entities_collector_->get_waitable(i);
+    if (waitable->is_ready(&wait_set_)) {
+      auto data = waitable->take_data();
+      waitable->execute(data);
```

Comment:
I'm wondering why we missed this in the first case, maybe you could think of a test to prevent this from regressing?

Reply:
I have no idea why we missed this in https://github.com/ros2/rclcpp/pull/1241.
Probably, the static single threaded executor is not being tested as much as the others (?).